Dtspb 567 wrap null exceptions

SarahLittlejohn opened the pull request:
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DTSPB-567


### Change description ###
Throw exception for null value.


**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X ] No
```

SarahLittlejohn committed:
DTSPB-567: Throw exception for null value for CaveatQueryService

SarahLittlejohn committed:
DTSPB-567: Removed the @Nullable

isha2040 commented on src/main/java/uk/gov/hmcts/probate/service/CaveatQueryService.java:
```diff
@@ -81,8 +84,7 @@ public CaveatData findCaveatById(CaseType caseType, String caveatId) {
     }
-
-    @Nullable
+    
```
can we please get ride of this new line and keep same as master

isha2040 committed:
DTSPB-567 enhancing the code

isha2040 committed:
DTSPB-567 removed the unused imports

isha2040 committed:
DTSSPB-567 Updated the failing condition

isha2040 committed:
DTSPB-567 removed an unused method from CaveatsQueryService
          removed any unused tests and added the new tests
          Added a new exception class in order to wrap

isha2040 committed:
DTSPB-567 Simplified the new Exception class
          removed unused imports

sanjayparekh-hmcts commented on src/main/java/uk/gov/hmcts/probate/service/CaveatQueryService.java:
```diff
@@ -106,14 +96,21 @@ public CaveatData findCaveatById(CaseType caseType, String caveatId) {
         return returnedCaveats.getCaveats();
     }
+
+    private static <T> T nonNull(@Nullable T result) {
```
do we really need @Nullable here?

isha2040 replied:
its just to indicate that the value passed could have null values , if you suggest to get rid of it can try that out as well

sanjayparekh-hmcts commented on src/main/java/uk/gov/hmcts/probate/exception/ClientDataException.java:
```diff
@@ -0,0 +1,38 @@
+        super(message);
+        this.errors = null;
+    }
+}
```
do you need a unit test on this class too?
